README.md

```diff
@@ -74,81 +121,184 @@ always ready.
 
+#### `zephyr-image` Update Module
+
+We provide the `zephyr-image` Update Module, which implements the update process for a Zephyr OS
+update integrated with MCUboot. Its [source code](core/src/mender-zephyr-image-update-module.c)
+can be inspected for inspiration and a better understanding of the expected behavior of each state.
+
+To use the `zephyr-image` Update Module, you need a board you need a board that supports
+MCUboot. The following link will filter the officially supported boards that also support MCUboot:
+* [Zephyr Project supported boards with MCU boot](https://docs.zephyrproject.org/latest/gsearch.html?q=MCUboot&check_keywords=yes&area=default#gsc.tab=0&gsc.q=MCUboot&gsc.ref=more%3Aboards&gsc.sort=)
+
 #### Update Modules State Machine
 
 As stated above, an Update Module for Mender MCU is set of customizable C functions. Concretely,
-once created, all Update Modules are identified by a C struct in
-[mender-zephyr-image-update-module.h](include/mender-zephyr-image-update-module.h). The
-most important part of the struct is the array of callbacks, one to be called for each state.
+once created, all Update Modules are identified by a C struct. The most important part of
+the struct is the array of callbacks, one to be called for each state.
 
 See [the state machine workflow
 diagram](https://docs.mender.io/artifact-creation/create-a-custom-update-module#the-state-machine-workflow)
 to learn about the flow between each state. An Update Module does not need to implement all of them,
 only the ones that are relevant for a particular type of update.
 
-We provide the `zephyr-image` Update Module, which implements the update process for a Zephyr OS
-update integrated with MCUboot. Its [source code](core/src/mender-zephyr-image-update-module.c)
-can be inspected for inspiration and a better understanding of the expected behavior of each state.
-
 After writing the code, you need to register the Update Module into the Mender MCU. See the register
 update module function in [`mender-client.h`](include/mender-client.h).
 
 
-## Experimental: testing the Mender MCU Client with POSIX
+### Network
 
-### Dependencies
-- CMake
-- libcurl
-- cJSON
+See this [example](https://github.com/mendersoftware/mender-mcu-integration/blob/main/src/utils/netup.c) from mender-mcu-integration for a demo.
+This implementation uses Zephyr's native networking APIs, see [Zephyr's Networking Guide](https://docs.zephyrproject.org/latest/connectivity/networking/index.html).
 
-Example for Ubuntu/Debian:
-```
-sudo apt install cmake libcurl4-openssl-dev libcjson-dev pkg-config
-```
-### Building the Client
+### Certificates
+See this [example](https://github.com/mendersoftware/mender-mcu-integration/blob/main/src/utils/certs.c) from mender-mcu-integration for a demo.
 
-1. Configure the build:
-```
-cmake -C cmake/CMake_posix_defaults.txt -B build tests/smoke
-```
+The `mender-mcu` client can take up to two certificates; typically one for the Server API calls, and
```

Comment:
Good you document this clearly but this seems like a limit we need to increase. If certificates need rotation we need to have the building blocks. I think we at least need support for 4 (two for each).

Reply:
This brings a point that we should rather move all the certs handling into mender-mcu core repo. Right now in the core we define if there will be one or two, but then is the application (mender-mcu-integration or any other user application) that will add the certs via `tls_credential_add(...)`. Instead, we can (and should?) have the mender client init function add the certs.

I suggest we create a task for this and do not modify the README further. In the task we shall write that the README needs to be updated accordingly.

Reply:
I don't quite see how this is part of the `mender-mcu` (client) scope. In my eyes, this is part of the "environment", just like the network setup and other similar things.

Reply:
That is a valid point (and the original point for having it out of Mender hands).

But it gets tricky for us when down in the code `zsock_setsockopt` needs to know the CA tags (up to 4).

At least we should think about how to simplify this part :thinking: We could maybe have yet another callback to get this `TLS_SEC_TAG_LIST` from the user.

To move the topic forward: discuss in next client team meeting and create a task either for moving it to mender-mcu (most likely not) or other ideas for decoupling it a bit (like mine above or others).